site/docs/v1/tech/tutorials/integrate-dotnet.md

```diff
@@ -0,0 +1,240 @@
+
+Now, copy and paste the following code into `Program.cs`.
+
+```csharp
+{% remote_include https://raw.githubusercontent.com/ritza-co/fusionauth-example-client-libraries/main/dotnet/Program.cs %}
+```
+
+Then, you’ll need to import a few NuGet packages:
+
+```shell
+dotnet add package JSON.Net # for debugging
+dotnet add package FusionAuth.Client # for our client access
+```
+
+You can then publish and run the application. Replace `<YOUR_API_KEY>` with the API key that you generated.
+
+```shell
+dotnet publish -r osx-x64
+fusionauth_api_key=<YOUR_API_KEY> bin/Debug/net7.0/osx-x64/publish/SetupFusionauth 
+```
+
+
+## Create Your {{page.technology}} Application
+
+Now you are going to create a {{page.technology}} application. While this section uses a simple {{page.technology}} application, you can use the same configuration to integrate your {{page.technology}} application with FusionAuth.
+
+You’ll use [Razor Pages](https://learn.microsoft.com/en-us/aspnet/core/razor-pages/?view=aspnetcore-7.0&tabs=visual-studio) and {{page.technology}}. This application will display common information to all users. There will also be a secured area, only available to an authenticated user. The [full source code](https://github.com/ritza-co/fusionauth-dotnet-integration) is available if you want to download it and take a look.
+
+First, create a new web application using the `dotnet` CLI and go to that directory:
+
+```shell
+dotnet new webapp -o SetupDotnet && cd SetupDotnet
+```
+
+To see the results, you should publish this application and run it. There are [multiple ways of deploying an application](https://docs.microsoft.com/en-us/dotnet/core/deploying/), but publishing ensures your deployment process is repeatable. You can use the [RID catalog](https://learn.microsoft.com/en-us/dotnet/core/rid-catalog) to build different versions of this application for different platforms. Here’s the command to publish a standalone executable you could deploy behind a proxy like Nginx:
+
+```shell
+dotnet publish -r osx-x64
+```
+
+Then start up the executable.
+
+```shell
+ASPNETCORE_ENVIRONMENT=Development bin/Debug/net7.0/osx-x64/publish/SetupDotnet
+```
+
+{% include _image.liquid src="/assets/img/docs/integrations/dotnet-integration/dotnet-welcome-page.png" alt="Home page for .Net app" class="img-fluid bottom-cropped" width="1200" figure=false %}
+
+You can hit `control-C` to exit this application.
+
+## Handle login for your {{page.technology}} application
+
+It’s always smart to leverage existing libraries as they are likely to be more secure and better handle edge cases. You’re going to add two new libraries to the application. Make sure you’re in the `SetupDotnet` directory and run these commands to add them.
+
+```shell
+dotnet add package Microsoft.AspNetCore.Authentication.OpenIdConnect
+dotnet add package IdentityModel.AspNetCore
```

Comment:
This is now an archived library: https://github.com/IdentityModel/IdentityModel.AspNetCore

Is there another option supported by microsoft that you can use?

Reply:
They moved development to a new repository: https://github.com/DuendeSoftware/Duende.AccessTokenManagement

Replacing `IdentityModel.AspNetCore` with `Duende.AccessTokenManagement.OpenIdConnect` and using `RevokeRefreshTokenAsync` instead of `RevokeUserRefreshTokenAsync` seems to work.